artifacts/reader.py

```diff
@@ -117,62 +165,53 @@ def _ReadArtifactDefinition(self, yaml_definition):
       if source_type:
         source_type.conditions = source.get(u'conditions', [])
         source_type.returned_types = source.get(u'returned_types', [])
-        self._ReadSupportedOS(yaml_definition, source_type, name)
+        self._ReadSupportedOS(source, source_type, name)
+        for supported_os in source_type.supported_os:
+          if supported_os not in artifact_object.supported_os:
```

Comment:
I was actually kind of curious about that myself.  The comments/documentation seem to indicate that that the intent is that supported_os, conditions, and returned_types are supposed to go away, but I would actually expect that supported_os and conditions should exist at the Source level.

Personally I would prefer to enforce that the top level supported_os match all of the defined sources, but I didn't want to do all that without understanding the intended direction.